File: typescript/0145-binary-tree-postorder-traversal.ts
function postorderTraversal(root: TreeNode | null): number[] {
    let res: number[] = [];

    function dfs(root) {
        if (!root) {
            return;
        }

        dfs(root.left);
        dfs(root.right);
        res.push(root.val);
    }

    dfs(root);
    return res;
}

File: c++/0145-binary-tree-postorder-traversal.cpp
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        if(!root) return {}; 

        postorderTraversal(root->left);
        postorderTraversal(root->right);
        result.push_back(root->val);
        
        return result;
    }
private:
    vector<int> result;
};